src/aliceVision/track/TracksBuilder.cpp

```diff
@@ -52,7 +73,99 @@ TracksBuilder::TracksBuilder()
+            const feature::EImageDescriberType descType = matchesIt.first;
+            const IndMatches& matches = matchesIt.second;
+
+            auto& featuresI = featuresPerDescI.at(descType);
+            auto& featuresJ = featuresPerDescJ.at(descType);
+            auto& duplicateFeaturesI = duplicateFeaturesPerDescI[descType];
+            auto& duplicateFeaturesJ = duplicateFeaturesPerDescJ[descType];
+
+            // per features match
+            for(const IndMatch& m : matches)
+            {
+                {
+                    auto& featureI = featuresI[m._i];
+                    auto& [duplicateFeatureIdsI, duplicateFeatureNodeI] =
+                        duplicateFeaturesI[DuplicateFeatureId(featureI.x(), featureI.y(), featureI.scale())];
+                    IndexedFeaturePair pairI(I, KeypointId(descType, m._i));
+                    auto& nodeI = map_indexToNode.at(pairI);
+                    if(duplicateFeatureNodeI == NULL)
+                    {
+                        duplicateFeatureIdsI.insert(m._i);
+                        duplicateFeatureNodeI = &nodeI;
+                    }
+                    else if(duplicateFeatureIdsI.insert(m._i).second)
+                    {
+                        _d->tracksUF->join(nodeI, *duplicateFeatureNodeI);
+                    }
+                }
+                {
+                    auto& featureJ = featuresJ[m._j];
+                    auto& [duplicateFeatureIdsJ, duplicateFeatureNodeJ] =
+                        duplicateFeaturesJ[DuplicateFeatureId(featureJ.x(), featureJ.y(), featureJ.scale())];
+                    IndexedFeaturePair pairJ(J, KeypointId(descType, m._j));
+                    auto& nodeJ = map_indexToNode.at(pairJ);
+                    if(duplicateFeatureNodeJ == NULL)
+                    {
+                        duplicateFeatureIdsJ.insert(m._j);
+                        duplicateFeatureNodeJ = &nodeJ;
+                    }
+                    else if(duplicateFeatureIdsJ.insert(m._j).second)
+                    {
+                        _d->tracksUF->join(nodeJ, *duplicateFeatureNodeJ);
+                    }
+                }
+            }
+        }
+    }
+
+    // fill duplicate features map
+    for(const auto& [viewId, duplicateFeaturesPerDesc] : duplicateFeaturesPerView)
+        for(const auto& [descType, duplicateFeatures] : duplicateFeaturesPerDesc)
+            for(const auto& [duplicateFeatureId, duplicateFeature] : duplicateFeatures)
+            {
+                auto& duplicateFeatureIdsSet = duplicateFeature.first;
+                size_t indexedFeaturePair_0 = *duplicateFeatureIdsSet.begin();
+                for(const auto& featureId : duplicateFeatureIdsSet)
+                {
+                    const auto& indexedFeaturePair_i = IndexedFeaturePair(viewId, KeypointId(descType, featureId));
+                    _duplicateFeaturesMap[indexedFeaturePair_i] = indexedFeaturePair_0;
+                }
+            }
```

Comment:
Doesn't it lack some elimination step ?

Reply:
Elimination of what?

If you mean elimination of merged tracks, there's no need to do it, the corresponding class ids are merged when merging the tracks, so any consequent use of the graph (notably using Union Find operations to iterate over found items/classes/tracks, e.g. `UnionFindEnum<IndexMap>::ItemIt`) will be able to correctly process the merged tracks as one unique track.

Reply:
I understand what you mean now. Yes that's correct, but it's been the case originally, that's why there's the `filter` function with `clearForks` option to remove these tracks.
Moreover, when using `exportToSTL`, only one observation per track is kept for one view, therefore the problems in question will not occur.

Reply:
@servantftechnicolor Is this remark resolved or is there something to be done?